### PR TITLE
doc: add interlinks between `echo` ←→ `printf` commands

### PR DESCRIPTION
**Pull request by Delapouite**
Hello

`echo` and `printf` are related functions which could benefit for more obvious links to navigate between them.

### CODE REVIEW

File: doc_src/cmds/echo.rst [hunk 1]
```diff
@@ -57,18 +57,20 @@ If ``-e`` is used, the following sequences are recognized:
...
 
     echo -e 'Top\\nBottom'
 
 Print Top and Bottom on separate lines, using an escape sequence
+
+
+See Also
+--------
+
+- :ref:`printf <cmd-printf>` command
```
Comment (faho) [inline doc_src/cmds/echo.rst]: I'd leave a short explanation as to why it's related or why you should look it up.

E.g. "the printf command, for more control over output formatting"?

File: doc_src/cmds/printf.rst [hunk 2]
```diff
@@ -66,18 +66,19 @@ This file has been imported from the printf in GNU Coreutils version 6.9. If you
...
 ::
 
     printf '%s: %d' "Number of bananas in my pocket" 42
 
 Will print "Number of bananas in my pocket: 42", _without_ a newline.
+
+See Also
+--------
+
+- :ref:`echo <cmd-echo>` command
```
Comment (faho) [inline doc_src/cmds/printf.rst]: "the echo command, for simpler output"

**Comment by Delapouite** [comment]
Thanks for your suggestions, I've adjusted the descriptions accordingly.

**Comment by faho** [comment]
Thank you, merged!

**Pull request merged**